jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/ChangeConnectionSettingsMenu.kt

```diff
@@ -44,26 +47,43 @@ class ChangeAccountSettingsActionGroup(private val project: Project, private val
             }
         }
 
-        val usedCredentials = accountSettingsManager.recentlyUsedCredentials()
-        if (usedCredentials.isEmpty()) {
-            actions.add(Separator.create(message("settings.credentials")))
+        if (mode.showCredentials) {
+            val usedCredentials = accountSettingsManager.recentlyUsedCredentials()
+            if (usedCredentials.isEmpty()) {
+                actions.add(Separator.create(message("settings.credentials")))
 
-            credentialSelector.isPopup = false
-            actions.add(credentialSelector)
-        } else {
-            actions.add(Separator.create(message("settings.credentials.recent")))
-            usedCredentials.forEach {
-                actions.add(ChangeCredentialsAction(it))
-            }
+                credentialSelector.isPopup = false
+                actions.add(credentialSelector)
+            } else {
+                actions.add(Separator.create(message("settings.credentials.recent")))
+                usedCredentials.forEach {
+                    actions.add(ChangeCredentialsAction(it))
+                }
 
-            credentialSelector.isPopup = true
-            actions.add(credentialSelector)
+                credentialSelector.isPopup = true
+                actions.add(credentialSelector)
+            }
         }
 
         CachedValueProvider.Result.create(actions.toTypedArray(), accountSettingsManager)
     }
 }
 
+enum class ChangeAccountSettingsMode(
+    internal val showRegions: Boolean,
+    internal val showCredentials: Boolean,
+    internal val nonSelected: String,
```

Comment:
Can we fold this and selectedDisplayFormat together, it seems are are mutually exclusive and that its usage implies `selectedDisplayFormat ?: nonSelected`